header/p2p/options.go

```diff
@@ -100,18 +108,45 @@ func DefaultClientParameters() *ClientParameters {
 		MinResponses:         2,
 		MaxRequestSize:       512,
 		MaxHeadersPerRequest: 64,
+		GC:                   time.Minute * 30,
+		MaxAwaitingTime:      time.Hour,
+		DefaultScore:         1,
+		MaxPeerTrackerSize:   100,
 	}
 }
 
+const (
+	greaterThenZero = "should be greater than 0"
+	providedSuffix  = "Provided value"
+)
+
 func (p *ClientParameters) Validate() error {
 	if p.MinResponses <= 0 {
-		return fmt.Errorf("invalid minimum amount of responses: %s", errSuffix)
+		return fmt.Errorf("invalid MinResponses: %s. %s: %v",
+			greaterThenZero, providedSuffix, p.MinResponses)
 	}
 	if p.MaxRequestSize == 0 {
-		return fmt.Errorf("invalid max request size: %s", errSuffix)
+		return fmt.Errorf("invalid MaxRequestSize: %s. %s: %v", greaterThenZero, providedSuffix, p.MaxRequestSize)
+	}
+	if p.MaxHeadersPerRequest == 0 {
+		return fmt.Errorf("invalid MaxHeadersPerRequest: %s. %s: %v", greaterThenZero, providedSuffix, p.MaxHeadersPerRequest)
+	}
+	if p.MaxHeadersPerRequest > p.MaxRequestSize {
+		return fmt.Errorf("MaxHeadersPerRequest should not be more than MaxRequestSize."+
+			"MaxHeadersPerRequest: %v, MaxRequestSize: %v", p.MaxHeadersPerRequest, p.MaxRequestSize)
 	}
-	if p.MaxHeadersPerRequest == 0 || p.MaxHeadersPerRequest > p.MaxRequestSize {
-		return fmt.Errorf("invalid max headers per request: %s", errSuffix)
+	if p.GC == 0 {
+		return fmt.Errorf("invalid gc period for peerTracker: %s. %s: %v", greaterThenZero, providedSuffix, p.GC)
+	}
+	if p.MaxAwaitingTime == 0 {
+		return fmt.Errorf("invalid MaxAwaitingTime for peerTracker: "+
+			"%s. %s: %v", greaterThenZero, providedSuffix, p.MaxAwaitingTime)
+	}
+	if p.DefaultScore < 0 {
+		return fmt.Errorf("invalid DefaultScore: %s. %s: %f", greaterThenZero, providedSuffix, p.DefaultScore)
+	}
```

Comment:
Small typo: Evaluation logic and error message mismatch if `DefaultScore` == 0.

Reply:
Nice catch. But lets make it `p.DefaultScore =< 0`